lib/marshalls/typosquatting.marshall.js

```diff
@@ -24,9 +24,9 @@ class Marshall extends BaseMarshall {
     let levenshteinDistance = null
     let similarPackages = []
     let packageFoundInTopPackages = false
-    return new Promise((resolve, reject) => {
+    return new Promise(async (resolve, reject) => {
       // If package is within an allow-list
-      if (this.packageRepoUtils.isPackageInAllowList(pkg.packageName)) {
+      if (await this.packageRepoUtils.isPackageInAllowList(pkg.packageName)) {
```

Comment:
Unsure how to fix the "anti-pattern" issue but feel free to push to this branch :)

Reply:
how about we change the function to just not be async, as there's no reason for it to be that:

```js
  async isPackageInAllowList(packageName) {
    const allowList = ['npq', 'ai', 'bun', 'deno']
    return allowList.includes(packageName)
  }
```

sounds like a better fix :-)